internal/middleware/logger.go

```diff
@@ -21,13 +21,15 @@ func LoggerMiddleware(rootLogger *zerolog.Logger) func(next http.Handler) http.H
 			bytesIn, _ := strconv.Atoi(r.Header.Get("Content-Length"))
 			traceId := ctxval.TraceId(r.Context())
 			edgeId := ctxval.EdgeRequestId(r.Context())
+			corrId := ctxval.CorrelationId(r.Context())
 			lctx := rootLogger.With().
 				Timestamp().
 				Str("trace_id", traceId).
 				Str("remote_ip", r.RemoteAddr).
 				Str("url", r.URL.Path).
 				Str("method", r.Method).
-				Int("bytes_in", bytesIn)
+				Int("bytes_in", bytesIn).
+				Str("correlation_id", corrId)
 			if edgeId != "" {
 				lctx = lctx.Str("request_id", edgeId)
 			}
```

Comment:
Can you treat the newly added field the same way as edgeId? Meaning, when it is not present (blank string) we would not add the empty field into the logger.